### PR TITLE
Firebase database

### PR DESCRIPTION
**Pull request by AndrewSakovich**

### CODE REVIEW

File: src/screens/AddNewItemScreen/index.tsx [hunk 1]
```diff
@@ -1,24 +1,43 @@
...
+      )
+      .ref(`Users/${userToken}/Todo/`)
+      .child(`${newItem.id}`)
+      .set(newItem);
+  };
 
+  const addItem = async (text: TodoItemType['text']) => {
+    const newItem = createNewItemHelper(text);
+    await setData(newItem, userToken);
     dispatch(addItemAction({ newItem }));
```
Comment (evgenyshenets91) [inline src/screens/AddNewItemScreen/index.tsx]: comment the code

File: src/screens/LoginScreen/index.tsx [hunk 2]
```diff
@@ -1,13 +1,58 @@
...
+    const userArr = firebase
+      .app()
+      .database(
+        'https://fir-2f0d3-default-rtdb.europe-west1.firebasedatabase.app/',
+      )
+      .ref('Users')
+      .on('value', snapshot => {
+        const obj = snapshot.val() ?? {};
+        const arr: SignInPayload['userToken'][] = Object.keys(obj);
+        console.log('ARR', arr);
```
Comment (evgenyshenets91) [inline src/screens/LoginScreen/index.tsx]: remove

Comment (AndrewSakovich) [reply src/screens/LoginScreen/index.tsx]: ok

File: src/screens/AddNewItemScreen/index.tsx (hunk 1 above)
Comment (AndrewSakovich) [reply src/screens/AddNewItemScreen/index.tsx]: I fix it

File: src/components/TodoItem/index.tsx [hunk 3]
```diff
@@ -1,36 +1,55 @@
...
+import { useSelector } from 'react-redux';
 import { TodoItemPropsType } from './types';
-import { doneItemAction } from '../../redux/actions/todoActions/doneItemAction';
+import { firebase } from '@react-native-firebase/database';
+import { ReduxStoreType } from '../../redux/store';
+import { SignInPayload } from '../../redux/actions/todoActions/signInAction';
+import { userTokenSelector } from '../../redux/selectors/userTokenSelector';
 
 export const TodoItem: FC<TodoItemPropsType> = props => {
+  const userToken = useSelector<ReduxStoreType, SignInPayload['userToken']>(
```
Comment (SHEMPEL133) [inline src/components/TodoItem/index.tsx:12]: destructure props on the first line of your component body
after useSelector and useState

File: src/components/TodoItem/index.tsx [hunk 4]
```diff
@@ -1,36 +1,55 @@
...
-    ]);
+  const onPressDelete = async () => {
+    await firebase
+      .app()
+      .database(
+        'https://fir-2f0d3-default-rtdb.europe-west1.firebasedatabase.app/',
+      )
+      .ref(`Users/${userToken}/Todo/${id}`)
+      .remove();
+    // Alert.alert('Delete task', 'Are you sure?', [
```
Comment (SHEMPEL133) [inline src/components/TodoItem/index.tsx:42]: why did you remove alert? put your firebase action to alert

File: src/screens/AddNewItemScreen/index.tsx [hunk 5]
```diff
@@ -1,25 +1,44 @@
...
   const [text, setText] = useState<string>('');
   const buttonStyle = text ? style.button : style.buttonDis;
 
-  const addItem = (text: TodoItemType['text']) => {
-    const newItem = createNewItemHelper(text);
+  const setData = async (
+    newItem: TodoItemType,
+    userToken: SignInPayload['userToken'],
+  ) => {
+    await firebase
```
Comment (SHEMPEL133) [inline src/screens/AddNewItemScreen/index.tsx:28]: these lines duplicate every time when you are calling firebase 

```
firebase
      .app()
      .database(
        `https://fir-2f0d3-default-rtdb.europe-west1.firebasedatabase.app/`,
      )
```

make sense to move it to service helper

File: src/screens/LoginScreen/index.tsx [hunk 6]
```diff
@@ -1,13 +1,55 @@
...
+      .app()
+      .database(
+        'https://fir-2f0d3-default-rtdb.europe-west1.firebasedatabase.app/',
+      )
+      .ref('Users')
+      .on('value', snapshot => {
+        const obj = snapshot.val() ?? {};
+        const arr: SignInPayload['userToken'][] = Object.keys(obj);
+        setUsers(arr);
+      });
```
Comment (SHEMPEL133) [inline src/screens/LoginScreen/index.tsx:28]: enter line before `return `

File: src/screens/UserInfoScreen/index.tsx [hunk 7]
```diff
@@ -9,10 +9,48 @@ import { TodoItemType } from '../../models';
...
+  const path = (userToken: SignInPayload['userToken']) => {
+    return `Users/${userToken}/Todo`;
+  };
+
+  const userInfo = useSelector<ReduxStoreType, TodoReducerState['user']>(
+    state => {
+      return state.user;
+    },
+  );
+  const userToken = userInfo?.uid || null;
```
Comment (SHEMPEL133) [inline src/screens/UserInfoScreen/index.tsx:28]: use nullish coalescing
